tools/sfxconvert.py

```diff
@@ -0,0 +1,176 @@
+    if(newId == 'INVALID_ID'):
+        print('ID parse error at line', i, 'in', func)
+        return -2
+    for k in range(i, j):
+        sourcedata[k] = sourcedata[k].replace(sfxId,newId)
+    if Verbose:
+        print('Replaced', sfxId, 'with', newId, 'in', func, 'at line', i + 1)
+    return 1
+
+def find_audio_func(line):
+    audiofuncs = list(AudioFunctions.keys())
+    for func in audiofuncs:
+        if(line.count(func)):
+            return func
+    return False
+
+def fix_sfx_all(repo):
+    global Verbose2
+
+    tv = Verbose2
+    Verbose2 = False
+
+    for subdir, dirs, files in os.walk(repo + os.sep + 'src'):
+        for filename in files:
+            if(filename.endswith('.c')):
+                file = subdir + os.sep + filename
+                fix_sfx(file, repo)
+
+    Verbose2 = tv
+    return 1
+
+def fix_sfx(file, repo, outfile = None):
+    if(outfile == None):
+        outfile = file
+    make_audio_dict(AudioFunctions, repo)
+    with open(file,'r',encoding='utf-8') as sourcefile:
+        sourcedata = sourcefile.readlines()
+    replacements = set()
+    j = 0
+    lookuperrors = 0
+    funcerrors = 0
+    iderrors = 0
+    fixes = 0
+    for i, line in enumerate(sourcedata):
+        if(i < j):
+            continue
+        if(find_audio_func(line)):
+            if(line.count(';')):
+                j = i + 1
+            else:
+                j = i
+                while(sourcedata[j].count(';') == 0):
+                    j += 1
+            status = fix_sfx_func(sourcedata, i, j + 1, repo)
+            if(status == -3):
+                funcerrors += 1
+            elif(status == -2):
+                iderrors += 1
+            elif(status == -1):
+                lookuperrors += 1
+            elif(status > 0):
+                fixes += 1
+    if(fixes > 0):
+        with open(outfile,'w',encoding ='utf-8',newline='\n') as outfile:
+            outfile.writelines(sourcedata)
+        if Verbose:
+            print(file, 'updated')
+    elif Verbose2:
+        print('No changes made to', file)
+    if(lookuperrors > 0):
+        print('Problem with function lookup. Try formatting functions.h')
+    if(funcerrors > 0):
+        print('Problem with function parsing. Encountering this message should be impossible, so please report that you did.')
+    if(iderrors > 0):
+        print('Problem with id parsing. Make sure your SFX ids are in hex.')
+    return 1
+
+parser = argparse.ArgumentParser(description='Convert hex SFX ids to macros')
+parser.add_argument('file', help="source file to be processed")
+parser.add_argument('repo', help="directory of the decomp repo")
+parser.add_argument('-o', metavar='outfile',dest='outfile',help='file to write to instead of original')
+parser.add_argument('-v', action='store_true',help='show what changes are made')
+
+# if __name__ == "__main__":
+#     args = parser.parse_args()
+#     set_verbose(args.v)
+#     fix_sfx(args.file, args.repo, outfile=args.outfile)
```

Comment:
Intended to have the main execution commented out?

Reply:
Oops, forgot to undo that after testing.